shared-libs/cht-script-api/test/auth.spec.js

```diff
@@ -0,0 +1,261 @@
+        can_edit: [ 'chw_supervisor' ],
+        can_configure: [ 'nurse' ]
+      };
+      const userRoles = [ 'chw_supervisor', 'gateway' ];
+
+      const result = auth.hasPermissions('can_configure', userRoles, chtPermissions);
+
+      expect(result).to.be.false;
+    });
+
+    it('should return true when user is admin', () => {
+      const chtPermissions = {
+        can_edit: [ 'chw_supervisor' ],
+        can_configure: [ 'nurse' ]
+      };
+      const userRoles = [ '_admin' ];
+
+      const result = auth.hasPermissions('can_create_people', userRoles, chtPermissions);
+
+      expect(result).to.be.true;
+    });
+
+    it('should return false when settings doesnt have roles assigned for the permission', () => {
+      const chtPermissions = {
+        can_edit: [ 'chw_supervisor' ],
+        can_configure: null
+      };
+      const userRoles = [ 'chw_supervisor' ];
+
+      const result = auth.hasPermissions('can_configure', userRoles, chtPermissions);
+
+      expect(result).to.be.false;
+    });
+
+    it('should return false for unknown permission', () => {
+      const chtPermissions = {
+        can_backup_facilities: [ 'national_admin' ],
+        can_export_messages: [ 'national_admin', 'district_admin', 'analytics' ]
+      };
+
+      const result = auth.hasPermissions([ 'xyz' ], [ 'district_admin' ], chtPermissions);
+
+      expect(result).to.be.false;
+    });
+
+    it('should return true for disallowed unknown permission', () => {
+      const chtPermissions = {
+        can_backup_facilities: [ 'national_admin' ],
+        can_export_messages: [ 'national_admin', 'district_admin', 'analytics' ]
+      };
+
+      const result = auth.hasPermissions([ '!xyz' ], [ 'district_admin' ], chtPermissions);
+
+      expect(result).to.be.true;
+    });
+
+    it('should return false when user is admin and has disallowed permission', () => {
+      const result = auth.hasPermissions([ '!can_backup_facilities' ], [ '_admin' ], []);
+
+      expect(result).to.be.false;
+    });
+
+    it('should rejects when user has one of disallowed permission', () => {
+      const chtPermissions = {
+        can_backup_facilities: [ 'national_admin' ],
+        can_export_messages: [ 'national_admin', 'district_admin', 'analytics' ]
+      };
+
+      const result = auth.hasPermissions(
+        [ '!can_backup_facilities', '!can_export_messages' ],
+        [ 'analytics' ],
+        chtPermissions
+      );
+
+      expect(result).to.be.false;
+    });
+
+    it('should return true when user doesnt have disallowed permissions', () => {
+      const chtPermissions = {
+        can_backup_facilities: [ 'national_admin' ],
+        can_export_messages: [ 'national_admin', 'district_admin', 'analytics' ]
+      };
+
+      const result = auth.hasPermissions(
+        [ '!can_backup_facilities', 'can_export_messages' ],
+        [ 'analytics' ],
+        chtPermissions
+      );
+
+      expect(result).to.be.true;
+    });
+  });
```

Comment:
The auth service spec has a few more test cases which have value to be added here, along with: 
- a test case with multiple permissions and multiple roles
- a test case with multiple permissions spread across roles
- a test case with an unknown disallowed permission
- a test case with no roles
- a test case with an empty permission (empty string) 
-

Reply:
Nice! I'm adding these

Reply:
Adding -> test case with multiple permissions and multiple roles

Reply:
Unit test added :)